src/map/battlefield.cpp

```diff
@@ -453,11 +540,13 @@ void CBattlefield::OpenChestinBcnm() {
 
 bool CBattlefield::loseBcnm() {
     beforeCleanup();
-    for (auto&& PChar : m_PlayerList)
+    for (int i = 0; i < m_PlayerList.size(); i++)
```

Comment:
I'm not entirely sure it is either, but it works. I hit this issue repeatedly when testing the lose containers on Karabara leaving the battlefield. An event would flood the player making the leave get interrupted an never happen. I got rid of the auto and went back to the old school style because I'm unfamiliar with modern C++ and kept getting a server crash related to the dynamic typing, and the rest of the file is using the old style anyways